lib/Lhx.hs

```diff
@@ -17,6 +17,7 @@ type Op = (Input -> Either Error Text)
 data Input = Input
   { iRaw    :: !Text
   , iFields :: [Text]
+  , iLength :: Int
```

Comment:
But why? Just `reverse iFields` and do `at` :)

Reply:
I just refactored `at` and now it supports negative indices. So you now don't need to modify it and `Input` should be kept intact.